plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -2239,6 +2239,16 @@ public ConnectorTableProperties getTableProperties(ConnectorSession session, Con
         HiveTableHandle handle = (HiveTableHandle) tableHandle;
         checkArgument(handle.getAnalyzePartitionValues().isEmpty() || constraint.getSummary().isAll(), "Analyze should not have a constraint");
 
+        if (!canPredicateFilterPartitions(constraint) && !constraint.getSummary().isNone() && handle.getPartitions().isEmpty()) {
```

Comment:
It it possible to test it (e.g with some super small limit)

Reply:
test added